proto/internal/temporal/server/api/deployment/v1/message.proto

```diff
@@ -0,0 +1,60 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package temporal.server.api.deployment.v1;
+
+option go_package = "go.temporal.io/server/api/deployment/v1;deployment";
+
+import "temporal/api/enums/v1/task_queue.proto";
+import "google/protobuf/timestamp.proto";
+
+message TaskQueue{
+    string name = 1;
+    temporal.api.enums.v1.TaskQueueType task_queue_type = 2;
+}
+
+message DeploymentTaskQueue{
+    google.protobuf.Timestamp first_poller_time = 1;
+    TaskQueue task_queue = 2;
+}
+
+message DeploymentWorkflowArgs {
+    string namespace_name = 1;
+    string namespace_id = 2;
+    repeated DeploymentTaskQueue deployment_task_queues = 3;
+}
```

Comment:
Thinking a couple steps ahead, we can structure this in the following way to avoid repeating TQ names in the payload and hence allow more TQs in a single deployment:
```suggestion
message DeploymentWorkflowArgs {
    string namespace_name = 1;
    string namespace_id = 2;
    map<string, TaskQueueFamilyInfo> task_queue_families = 3;
    
    message TaskQueueFamilyInfo {
        repeated TaskQueueInfo task_queues = 1;
        
        message TaskQueueInfo {
            temporal.api.enums.v1.TaskQueueType task_queue_type = 1;
            google.protobuf.Timestamp first_poller_time = 2;
        }
    }
}
```